docs/api/IoT.js-API-File-System.md

```diff
@@ -32,196 +30,273 @@ The following shows fs module APIs available for each platform.
+        - [`fs.writeSync(fd, buffer, offset, length, position)`](#fswritesyncfd-buffer-offset-length-position)
+        - [`fs.writeFile(path, data, [, options], callback)`](#fswritefilepath-data--options-callback)
+        - [`fs.writeFileSync(path, data, [, options])`](#fswritefilesyncpath-data--options)
+        - [`fs.unlink(path, callback)`](#fsunlinkpath-callback)
+        - [`fs.unlinkSync(path)`](#fsunlinksyncpath)
+        - [`fs.close(fd, callback)`](#fsclosefd-callback)
+        - [`fs.closeSync(fd)`](#fsclosesyncfd)
+    - [Class: fs.Stats](#class-fsstats)
+        - [Prototype Functions](#prototype-functions)
+            - [`stats.isDirectory()`](#statsisdirectory)
+            - [`stats.isFile()`](#statsisfile)
+
+
+# File System
+
+
+## Module Functions
+
+### `fs.open(path, flags[, mode], callback)`
+* `path <String>` - file path to be opened.
+* `flags <String>` - open flags.
+* `mode <Number>` Default: `0666` - permission mode.
+* `callback <Function(err: Error | null, fd: Number)>`
+
+Opens file asynchronously.
+
+**Example**
+
+```js
+var fs = require('fs');
+fs.open('test.txt', 'r', 755, function(err, fd) {
+  if (err) {
+    throw err;
+  }
+});
+```
+
+### `fs.openSync(path, flags[, mode])`
+* `path <String>` - file path to be opened.
+* `flags <String>` - open flags.
+* `mode <Number>` Default: `0666` - permission mode.
+
+Opens file synchronously.
+
+
+### `fs.read(fd, buffer, offset, length, position, callback)`
+* `fd <Integer>` - file descriptor.
+* `buffer <Buffer>` - buffer that the data will be written to.
+* `offset <Number>` - offset of the buffer where to start writing.
+* `length <Number>` - number of bytes to read.
+* `position <Number>` - specifying where to start read data from the file, if `null`, read from current position.
+* `callback <Function(err: null | Error, bytesRead: Number, buffer: Buffer)>`
+
+Reads data from the file specified by fd asynchronously.
+
+**Example**
+
+```js
+var fs = require('fs');
+fs.open('test.txt', 'r', 755, function(err, fd) {
+  if (err) {
+    throw err;
+  }
+  var buffer = new Buffer(64);
+  fs.read(fd, buffer, 0, buffer.length, 0, function(err, bytesRead, buffer) {
+      if (err) {
+        throw err;
+      }
+    });
+});
+```
+
+
+### `fs.readSync(fd, buffer, offset, length, position)`
+* `fd <Integer>` - file descriptor.
+* `buffer <Buffer>` - buffer that the data will be written to.
+* `offset <Number>` - offset of the buffer where to start writing.
+* `length <Number>` - number of bytes to read.
+* `position <Number>` - specifying where to start read data from the file, if `null`, read from current position.
+
+Reads data from the file specified by fd synchronously.
+
+
+### `fs.readFile(path[, options], callback)`
+* `path <String>` - file path to be opened.
+* `options <Object>` - options for the operation.
+  * `encoding <String>` Default: `null` - encoding of the file.
```

Comment:
This is absurd. Default says there can be `null`  but type signature only allows to come as `String`